File: Other Research/Python ffti Algorithm/v5/rho_construct_test.py
import ffti_v6 as fi
import numpy as np
from time import time

x_max, y_max, z_max = 2 * np.pi, 2 * np.pi, 2 * np.pi
Ni, Nj, Nk = 21, 21, 21

x, y, z = np.linspace(0, x_max, Ni), np.linspace(0, y_max, Nj), np.linspace(0, z_max, Nk)

for i in range(len(x)):
   print(x[i])

rho = np.zeros(shape = (Ni, Nj, Nk))

def func(x, y, z):
   rho = np.e ** (np.sin(x) + np.sin(y) + np.sin(z))
   return rho

for i in range(0, Ni): 
   for j in range(0, Nj): 
      for k in range(0, Nk):
         rho[i, j, k] = func(x[i], y[j], z[k])
         print("rho[{0}][{1}][{2}] = {3}".format(i, j, k, rho[i, j, k]))

File: Other Research/Dead End Projects/C ffti Development/rho_construct_test.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

double *linspace(double start, double stop, int n) {
   double *array = malloc(n * sizeof *array);
   for (int i = 0; i < n; i++) {
      array[i] = start + i * (stop - start) / (n - 1);
   }
   return array;
}

int main() {
   const int Ni = 21, Nj = 21, Nk = 21;
   const double x_max = 2 * M_PI, y_max = 2 * M_PI, z_max = 2 * M_PI;

   double *x = linspace(0, x_max, Ni);
   double *y = linspace(0, y_max, Nj);
   double *z = linspace(0, z_max, Nk);

   // for (int i = 0; i < Ni; i++) {
   //    printf("%f\n", x[i]);
   // }

   double rho[Ni][Nj][Nk];

   for (int i = 0; i < Ni; i++) {
      for (int j = 0; j < Nj; j++) {
         for (int k = 0; k < Nk; k++) {
            rho[i][j][k] = pow(M_E, sin(x[i]) + sin(y[j]) + sin(z[k]));
            printf("rho[%d][%d][%d] = %f\n", i, j, k, rho[i][j][k]);
         }
      }
   }
}